### ISSUE TITLE
ol.coordinate.toStringHDMS() can result in 60 seconds in the output string rather than rolling that up to the next minute

### ISSUE THREAD

**Issue by rjackson64840** [openlayers/openlayers#5827]
We are seeing a value like this being output from toStringHDMS():

37 59' 60" N ...

Which should actually be:

38 00' 00" N ...

I suspect it's due to expected floating point arithmetic errors and could be handled in degreesToStringHDMS_() like so:

```
ol.coordinate.degreesToStringHDMS_ = function(degrees, hemispheres, opt_fractionDigits) {
    var normalizedDegrees = ol.math.modulo(degrees + 180, 360) - 180;
    var x = Math.abs(3600 * normalizedDegrees);
    var dflPrecision = opt_fractionDigits || 0;

    var deg = Math.floor(x / 3600);
    var min = Math.floor((x / 60) % 60);
    var precision = Math.pow(10, dflPrecision);
    var sec = Math.ceil((x * precision) % 60) / precision;

    if (60 === sec) {
        sec = 0;
        min += 1;
    }

    if (min >= 60) {
        min -= 60;
        deg = deg >= 0 ? deg + 1 : deg - 1;
    }

    return deg + '\u00b0 ' + ol.string.padNumber(min, 2) + '\u2032 ' +
        ol.string.padNumber(sec, 2, dflPrecision) + '\u2033 ' +
        hemispheres.charAt(normalizedDegrees < 0 ? 1 : 0);
};

```

**Comment by marcjansen** [comment]
Do you have a concrete failing testcase? Are you willing and able to submit a pull request?

**Comment by rjackson64840** [comment]
Thanks.  I can come up with a test case and would be happy to submit a pull request (don't know how soon though).

**Comment by marcjansen** [comment]
No rush.

**Comment by veetrp** [comment]
Noticed the above as well. I solved it by following function operating on a either a lat or a long. The function is comparable to the one proposed above. Precision is up to 9 digits in addition to the total seconds. .lpad is my own extension for String for left padding. Anyway, my 2 cents for your perusal... Rgds, Reinoud

```
function toDegMinSec(d,decimals,isSigned,isLatitude) {
	var negative = d<0? true : false
	
	var sectotal = Math.abs(d*3600);
	decimals = parseInt(Math.max(0,Math.min(9,decimals)));
	var secrounded = sectotal.toFixed(decimals);
	
	var minint = parseInt(secrounded/60.0);
	var sec = (secrounded-(60*minint)).toFixed(decimals);
	var min = parseInt(minint%60);
	var deg = parseInt(minint/60);
	
	if (isSigned) {
		var sign = deg==0? " " : negative? "-" : " ";
		return (sign+new String(deg).lpad(' ',3)+"°"+new String(min).lpad(' ',2)+"'"+new String(sec).lpad(' ',(2+1+decimals))+"\"");
	} else {
		var hem = d<0? (isLatitude? "S":"W") : isLatitude? "N" : "E";
		return (new String(deg).lpad(' ',3)+"°"+new String(min).lpad(' ',2)+"'"+new String(sec).lpad(' ',(2+1+decimals))+"\""+hem);
	}
}

```

**Comment by ahocevar** [comment]
Pull request @veetrp?

**Comment by veetrp** [comment]
I'm sorry ahocevar. I'm new to GitHub and not well versed yet in this kind of collaboration. I would love to participate but forking the whole ol3 lib is probably a bit much to begin with... Please indicate if my perception is wrong... Cheers, Reinoud

**Comment by veetrp** [comment]
@ahocevar referring to the above, making a pull request means I have to fork ol and add my function to my branch and then request a pull? Or is there an easier way? Sorry I'm a newbie on GitHub and I code in my spare time (being a geologist by profession)...

**Comment by tsauerwein** [comment]
@veetrp This tutorial might help you getting started: https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github

**Comment by veetrp** [comment]
@tsauerwein and @ahocevar 
Hi Tobias and Andreas, thanks for the info. As I suspected, to make a pull request you'll have to fork and branch. I appreciate that's the version control system, i.e. fair enough. However at this point that would take too much of my time, being only a developer in my spare time. So, I'm sorry. Anyways, I greatly enjoy this ol work and thank you for the all the hard work. Rgds, Reinoud

**Comment by rjackson64840** [comment]
#6410 I created a pull request to address this

**Comment by ahocevar** [comment]
Fixes with #6410.
